tortoise/tortoise_test.go

```diff
@@ -1351,32 +1353,33 @@ func TestSumVotes(t *testing.T) {
 			tortoise.trtl.atxdb = atxdb
 			consensus := tortoise.trtl
 
-			layers := [][]*types.Ballot{}
+			layers := [][]*types.Block{}
 			for i, layer := range tc.layers {
+				layerBlocks := []*types.Block{}
 				layerBallots := []*types.Ballot{}
 				lid := genesis.Add(uint32(i) + 1)
 				for j, b := range layer {
-					ballot := &types.Ballot{
-						InnerBallot: types.InnerBallot{
-							EligibilityProof: types.VotingEligibilityProof{J: uint32(j)},
-							AtxID:            activeset[b.ATX],
-							EpochData:        &types.EpochData{ActiveSet: activeset},
-							LayerIndex:       lid,
-						},
-					}
+					block := &types.Block{}
+					block.EligibilityProof = types.BlockEligibilityProof{J: uint32(j)}
+					block.ATXID = activeset[b.ATX]
+					block.ActiveSet = &activeset
+					block.LayerIndex = lid
 					// don't vote on genesis for simplicity,
 					// since we don't care about block goodness in this test
 					if i > 0 {
-						ballot.ForDiff = getDiff(layers, b.Support)
-						ballot.AgainstDiff = getDiff(layers, b.Against)
-						ballot.NeutralDiff = getDiff(layers, b.Abstain)
-						ballot.BaseBallot = layers[b.Base[0]][b.Base[1]].ID()
+						block.ForDiff = getDiff(layers, b.Support)
+						block.AgainstDiff = getDiff(layers, b.Against)
+						block.NeutralDiff = getDiff(layers, b.Abstain)
+						block.BaseBlock = layers[b.Base[0]][b.Base[1]].ID()
 					}
-					ballot.Initialize()
-					layerBallots = append(layerBallots, ballot)
+					block.Signature = signer.Sign(block.Bytes())
+					block.Initialize()
+					layerBlocks = append(layerBlocks, block)
+					layerBallots = append(layerBallots, block.ToBallot())
 				}
-				layers = append(layers, layerBallots)
+				layers = append(layers, layerBlocks)
 
+				consensus.processBlocks(ctx, layerBlocks)
```

Comment:
i think this change made the test more confusing.

we are testing `sumVotesForBlock()` here.
we should call `processBlocks()` on the blocks in the diff list, not on the ballots.

Reply:
getDiff is just a helper to compute votes from indexes, what is confusing?

Reply:
ok i see what you mean, i think because blocks and ballots are aliases in current codebase i made it this way

Reply:
but something is wrong with the test if you need to change the code for it to pass.
theoretically the setup should expect `t.BlockLayer` to only contains BlockIDs in the diff lists

Reply:
i will split up ballots and blocks in test